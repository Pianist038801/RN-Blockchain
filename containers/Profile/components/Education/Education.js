import React from 'react'
import PropTypes from 'prop-types'
import { View, Text } from 'react-native'
import VerifiedIcon from '../VerifiedIcon'
import { getDates } from '../../utils/date.js'
import styles from '../../../../style.js'

const Education = (props) => {
  const { education } = props
  var dates = getDates(education.start_date, education.end_date)
  const textStyle = date => date === 'Present' ?
    [styles.profileEntityDate, styles.profilePresentText] :
    styles.profileEntityDate

  return (
    <View style={styles.profileEntityContainer}>
      <VerifiedIcon type="education" entity={education} />
      <View style={styles.profileEntityContent}>
        <Text style={styles.profileEntityPrimaryText}>{education.studied}</Text>
        <Text>{education.institution}</Text>
      </View>
      <View style={styles.profileEntityRight}>
        <Text style={textStyle(dates.start_date)}>{ dates.start_date }</Text>
        <Text style={textStyle(dates.end_date)}>- { dates.end_date }</Text>
      </View>
    </View>
  )
}

Education.propTypes = {
  education: PropTypes.shape({
    start_date: PropTypes.string.isRequired,
    end_date: PropTypes.string.isRequired,
    institution: PropTypes.string.isRequired,
    studied: PropTypes.string.isRequired
  })
}

export default Education
