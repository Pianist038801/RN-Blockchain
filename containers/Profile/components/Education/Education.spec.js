import React from 'react'
import renderer from 'react-test-renderer'
import Education from './index'

describe('containers/Profile/components/Education', () => {
  it('renders correctly', () => {
    const props = {
      education: {
        institution: 'Westminter Uni',
        start_date: '2003-09-04',
        end_date: '2004-08-31',
        studied: 'MSc in Computer Science',
        verified: true
      }
    }

    const tree = renderer.create(
      <Education {...props} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
