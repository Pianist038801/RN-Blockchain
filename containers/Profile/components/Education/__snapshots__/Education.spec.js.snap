// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/Education renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#fff",
      "borderWidth": 1,
      "flex": 0.5,
      "flexDirection": "row",
      "marginBottom": 5,
      "padding": 22.5,
      "width": 600,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 75,
        "resizeMode": "contain",
        "width": 75,
      }
    }
  />
  <View
    style={
      Object {
        "marginLeft": 22.5,
        "width": 360,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#569cee",
          "fontWeight": "bold",
        }
      }
    >
      MSc in Computer Science
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Westminter Uni
    </Text>
  </View>
  <View
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
      2003
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
      - 
      2004
    </Text>
  </View>
</View>
`;
